Framework/Algorithms/test/FilterEventsTest.h

```diff
@@ -736,27 +732,31 @@ class FilterEventsTest : public CxxTest::TestSuite {
    * nanosecond
     * @param pulsedt : pulse length in int64_t format with unit nanosecond
     */
-  DataObjects::EventWorkspace_sptr
-  createEventWorkspaceInDirect(int64_t runstart_i64, int64_t pulsedt) {
+  EventWorkspace_sptr createEventWorkspaceInDirect(int64_t runstart_i64,
+                                                   int64_t pulsedt) {
     // Create an EventWorkspace with 10 banks with 1 detector each.  No events
     // is generated
-    DataObjects::EventWorkspace_sptr eventWS =
+    EventWorkspace_sptr eventWS =
         WorkspaceCreationHelper::createEventWorkspaceWithFullInstrument(10, 1,
                                                                         true);
-
     // Add EFixed to each detector
     const ParameterMap &pmap = eventWS->constInstrumentParameters();
 
     for (size_t i = 0; i < 10; ++i) {
-      Geometry::IDetector_const_sptr det = eventWS->getDetector(i);
-      Parameter_sptr par = pmap.getRecursive(det.get(), "Efixed");
+
+      const auto &spectrumInfo = eventWS->spectrumInfo();
+
+      if (!spectrumInfo.hasUniqueDetector(i))
+        continue;
+      const auto &det = spectrumInfo.detector(i);
+      Parameter_sptr par = pmap.getRecursive(&det, "Efixed");
       if (par) {
         // No need to set up E-Fix
         // double efix = par->value<double>();
         ;
       } else {
 
-        eventWS->setEFixed(det->getID(), 2.08);
+        eventWS->setEFixed(det.getID(), 2.08);
```

Comment:
Yes, this is the problem: Internally this obtains a non-const reference to the parameter map, I wonder if there is a nice way to fix this... :(